### PR TITLE
feat: get and patch pool model endpoints

### PR DESCRIPTION
**Pull request by TheRealSeber**
please, proceed to make a review, i need to know whether i am feeling mongodb the "right way"

### CODE REVIEW

File: app/api/routers/pool_model.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+"""
+This module contains the API routes and their corresponding
+functions for handling pool-model-related requests.
+"""
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
```
Comment (HermanPlay) [inline app/api/routers/pool_model.py:6]: This is obsolete

File: app/api/routers/pool_model.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+router = APIRouter(
+    prefix="/pool", tags=["pool-model"], dependencies=[Depends(JWTBearer())]
+)
+
+
+@router.patch("/{pool_id}/details", response_model=pool_schemas.Pool, status_code=200)
+async def patch_pool_model(
+    pool_id: int,
+    details: pool_model_schemas.PoolModelPatch,
+    db: Session = Depends(get_db),
```
Comment (HermanPlay) [inline app/api/routers/pool_model.py:22]: The type of the DB is not Session

File: app/api/routers/pool_model.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        )
+    return await PoolModelService.patch_pool_models(
+        db=db, pool_id=pool_id, details=details
+    )
+
+
+@router.get("/{pool_id}/details", response_model=pool_schemas.Pool, status_code=200)
+async def get_pool_models(
+    pool_id: int,
+    db: Session = Depends(get_db),
```
Comment (HermanPlay) [inline app/api/routers/pool_model.py:63]: Again, db type is not Session

File: app/api/routers/pool_model.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            detail=f"Some models are not found!\n\
+                Target models id's: [{', '.join(non_existing_target_models)}]\n\
+                Shadows models id's: [{', '.join(non_existing_shadow_models)}]",
+        )
+    return await PoolModelService.patch_pool_models(
+        db=db, pool_id=pool_id, details=details
+    )
+
+
+@router.get("/{pool_id}/details", response_model=pool_schemas.Pool, status_code=200)
```
Comment (HermanPlay) [inline app/api/routers/pool_model.py:60]: I think response model is incorrect

**Review by HermanPlay — CHANGES_REQUESTED** [review]
Some mistakes seen without testing, fix them. Besides that looks good

File: app/api/routers/pool_model.py (hunk 1 above)
Comment (TheRealSeber) [reply app/api/routers/pool_model.py:6]: What a sophisticated word mr F

Comment (TheRealSeber) [reply app/api/routers/pool_model.py:6]: You have to be indefatigable in that Mr.F !!!

File: app/api/schemas/pool_model.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+class PoolModelGet(BaseSchema):
+    targets: list[TargetsGet] = Field([], description="Routing targets")
+    shadows: list[ShadowsGet] = Field([], description="Routing shadow")
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "targets": [
+                    {
+                        "model_id": 1,
```
Comment (matined) [inline app/api/schemas/pool_model.py:43]: model_id is a string

**Review by matined — APPROVED** [review]

**Review by matined — COMMENTED** [review]
I wonder if we shouldn't change everything from PoolModel to PoolDetails etc

File: app/api/services/pool_model.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            )
+            shadow_models_return.append(
+                pool_model_schemas.ShadowsGet(
+                    **model_data.dict(),
+                    model_id=str(model_data.id),
+                )
+            )
+
+        print(f"DEBUG 1: {target_models_return}")
+        print(f"DEBUG 2: {shadow_models_return}")
```
Comment (HermanPlay) [inline app/api/services/pool_model.py:64]: You can remove this, or use logger

**Review by HermanPlay — CHANGES_REQUESTED** [review]
Small cleanup needed

**Pull request merged**